compiler/passes/normalize.cpp

```diff
@@ -386,6 +390,39 @@ static bool isInLifetimeClause(CallExpr* call) {
   return (fn && parentBlock && fn->lifetimeConstraints == parentBlock);
 }
 
+/************************************* | **************************************
+* Move each 'implements' constraint in a where clause, ex.
+*   proc constrainedGenericFun(...) where implements MyIFC(T1,T2) {...}
+* to the enclosing FnSymbol's interfaceConstraints list.
+*
+* TODO handle the case where the actuals of the implements constraints
+* undergo normalization.
+*
+* TODO handle the case where the 'where' clause has other things
+* besides a single IC, for example 'where IFC1(...) && IFC2(...)'.
+*
+************************************** | *************************************/
+
+static void moveInterfaceConstraints() {
+  forv_Vec(ImplementsExpr, ie, gImplementsExprs) {
+    if (isImplementsStmt(ie->parentExpr))
```

Comment:
Should this check parent's parent etc? In other words, is it possible that an ImplementsStmt contains an ImplementsExpr at some deeper nesting?